src/core/vectortile/qgsmapboxglstyleconverter.cpp

```diff
@@ -0,0 +1,1381 @@
+        break;
+    }
+  }
+
+  // buffer color
+  QColor bufferColor;
+  if ( jsonPaint.contains( QStringLiteral( "text-halo-color" ) ) )
+  {
+    const QVariant jsonBufferColor = jsonPaint.value( QStringLiteral( "text-halo-color" ) );
+    switch ( jsonBufferColor.type() )
+    {
+      case QVariant::Map:
+        ddLabelProperties.setProperty( QgsPalLayerSettings::BufferColor, parseInterpolateColorByZoom( jsonBufferColor.toMap(), context, &bufferColor ) );
+        break;
+
+      case QVariant::List:
+      case QVariant::StringList:
+        ddLabelProperties.setProperty( QgsPalLayerSettings::BufferColor, parseInterpolateListByZoom( jsonBufferColor.toList(), PropertyType::Color, context, 1, &bufferColor ) );
+        break;
+
+      case QVariant::String:
+        bufferColor = parseColor( jsonBufferColor.toString(), context );
+        break;
+
+      default:
+        context.pushWarning( QObject::tr( "Skipping non-implemented text-halo-color expression" ) );
+        break;
+    }
+  }
+
+  double bufferSize = 0.0;
+  if ( jsonPaint.contains( QStringLiteral( "text-halo-width" ) ) )
+  {
+    const QVariant jsonHaloWidth = jsonPaint.value( QStringLiteral( "text-halo-width" ) );
+    switch ( jsonHaloWidth.type() )
+    {
+      case QVariant::Int:
+      case QVariant::Double:
+        bufferSize = jsonHaloWidth.toDouble();
+        break;
+
+      case QVariant::Map:
+        bufferSize = 1;
+        ddLabelProperties.setProperty( QgsPalLayerSettings::BufferSize, parseInterpolateByZoom( jsonHaloWidth.toMap(), context ) );
+        break;
+
+      case QVariant::List:
+      case QVariant::StringList:
+        bufferSize = 1;
+        ddLabelProperties.setProperty( QgsPalLayerSettings::BufferSize, parseInterpolateListByZoom( jsonHaloWidth.toList(), PropertyType::Numeric, context ) );
+        break;
+
+      default:
+        context.pushWarning( QObject::tr( "Skipping non-implemented text-halo-width expression" ) );
+        break;
+    }
+  }
+
+  // TODO implement halo blur
+#if 0
+  if ( 'text-halo-blur' in json_paint )
+  {
+    json_text_halo_blur = json_paint['text-halo-blur'];
+    if ( isinstance( json_text_halo_blur, ( float, int ) ) )
+      buffer_size = buffer_size - json_text_halo_blur;
+    else
+      print( "skipping non-float text-halo-blur", json_text_halo_blur );
+  }
+#endif
+
+  QgsTextFormat format;
+  format.setSizeUnit( QgsUnitTypes::RenderPixels );
+  if ( textColor.isValid() )
+    format.setColor( textColor );
+  if ( textSize >= 0 )
+    format.setSize( textSize );
+  if ( foundFont )
+    format.setFont( textFont );
+
+  if ( bufferSize > 0 )
+  {
+    format.buffer().setEnabled( true );
+    format.buffer().setSize( bufferSize * PIXEL_RATIO );
+    format.buffer().setSizeUnit( QgsUnitTypes::RenderPixels );
+    format.buffer().setColor( bufferColor );
+  }
+
+  QgsPalLayerSettings labelSettings;
+
+  // TODO - likely improvements required here
+  labelSettings.fieldName = QStringLiteral( "name:latin" );
```

Comment:
I think we don't need `name:latin` as the default field name - in the original code I used it just because I did not parse field name at all and this field was used in OpenMapTiles styles...

Reply:
Thanks -- I've gone back and revisited this text-field handling, it's much more robust now!